presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java

```diff
@@ -41,29 +41,16 @@ public SchemaTablePrefix(String schemaName, String tableName)
         this.tableName = Optional.of(checkNotEmpty(tableName, "tableName"));
     }
 
-    /**
-     * @deprecated replaced by {@link #getSchema()}
-     */
-    @Deprecated
-    public String getSchemaName()
+    public boolean isEmpty()
```

Comment:
Move this down to be before the `matches` method to make the diff easier to read

Reply:
Also change `matches()` to call this method for the first check

Reply:
Maybe I read the diff wrong originally, but I think I was suggesting this:
```java
public boolean matches(SchemaTableName schemaTableName)
{
    // empty prefix matches everything
    if (isEmpty()) {
        return true;
    }

    if (!schemaName.get().equals(schemaTableName.getSchemaName())) {
        return false;
    }

    return !tableName.isPresent() || tableName.get().equals(schemaTableName.getTableName());
}
```
I agree that the `toOptionalSchemaTableName()` way is hard to read.